### DISCUSSION TITLE
How can I bind coc function from within my own function (used for insert mode)

### DISCUSSION THREAD

**Discussion by reteehee** [neoclide/coc.nvim#3629] [Q&A] [answered]
How can I call a coc function like `coc#refresh()` or `coc#_select_confirm()` from within my own function (which used for insert mode):

For example I have this:

``` vim
" Use <tab> to trigger completion
fun! s:coc_comp_tab() abort
    if pumvisible()
        return "\<c-n>"
    " If the next char is closed pairs, jump
    elseif getline('.')[col('.') - 1] =~ "['\"`)\\]}>]"
        return "\<right>"
    " If the previous char is space, tab
    elseif getline('.')[col('.') - 2] =~# '\s'
        return "\<tab>"
    else
        " Fucking try and error
        coc#refresh()
        " call coc#refresh()
        " return "\<c-o>:call coc#refresh()\<cr>"
    endif
endfun

inore <silent><expr> <tab> <sid>coc_comp_tab()
```

Please see the `else` clause above, that must be an error or throw unexpected result (in my case, it typed `0` when there's no popup completion window).

I want it to behave like the official example, when there's no condition meets, just do `coc#refresh()`:

``` vim
inoremap <silent><expr> <TAB>
      \ pumvisible() ? "\<C-n>" :
      \ <SID>check_back_space() ? "\<TAB>" :
      \ coc#refresh()
```

**Comment by chemzqm** [comment, marked as answer]
``` vim
function! coc#refresh() abort
  return "\<c-r>=coc#start()\<CR>"
endfunction
```
You should return the expression the same as `coc#refresh()`

**Comment by reteehee** [comment]
Where's `coc#start()` come from, if its equivalent to `coc#refresh()`, what about the equivalent for the others like `coc#_select_confirm`. Seems like there's no documentation.

 Also another function that doesn't have documentation like `coc#rpc#request()`

**Reply by chemzqm** [reply]
They're supposed to be internal, you can simply use `return coc#refresh()`
